src/DiffSharp.Core/Tensor.fs

```diff
@@ -222,55 +222,55 @@ type Tensor =
         if a.Shape = b.Shape then
             let inline fRaw(a:RawTensor,b) = a.AddTT(b)
             let inline fTensor(a,b) = a + b
-            let inline dfTensorFwdTT(cp,ap,ad,bp,bd) = ad + bd
-            let inline dfTensorFwdTC(cp,ap,ad) = ad
-            let inline dfTensorFwdCT(cp,bp,bd) = bd
+            let inline dfTensorFwdTT(_cp,_ap,ad,_bp,bd) = ad + bd

```

Comment:
@gbaydin These changes are needed to suppress the unused variable warnings.  I think they are clarifying as they capture the intent that various parts of the derivative depend on the inputs, and you have to explicitly mark up when they don't.